packages/main/src/plugin/index.ts

```diff
@@ -1037,9 +1041,20 @@ export class PluginSystem {
         imageName: string,
         callbackId: number,
       ): Promise<void> => {
-        return containerProviderRegistry.pullImage(providerContainerConnectionInfo, imageName, (event: PullEvent) => {
-          this.getWebContentsSender().send('container-provider-registry:pullImage-onData', callbackId, event);
-        });
+        return progressImpl.withProgress<void>(
+          { location: ProgressLocation.TASK_WIDGET, title: `Pulling ${imageName}.` },
+          async progress => {
+            await containerProviderRegistry.pullImage(
```

Comment:
There should be try/catch block or failure to pull will leave task running forever

![image](https://github.com/containers/podman-desktop/assets/620330/a884ab04-8522-437c-8b97-72b602f7c678)